backend_new/src/services/listing-csv-export.service.ts

```diff
@@ -0,0 +1,864 @@
+      },
+      {
+        path: 'events',
+        label: 'Open House',
+        format: (val: ListingEvent[]): string => {
+          if (!val) return '';
+          return val
+            .filter((event) => event.type === ListingEventsTypeEnum.openHouse)
+            .map((event) => {
+              let openHouseStr = '';
+              if (event.label) openHouseStr += `${event.label}`;
+              if (event.startTime) {
+                const date = formatLocalDate(
+                  event.startTime,
+                  'MM-DD-YYYY',
+                  this.timeZone,
+                );
+                openHouseStr += `: ${date}`;
+                if (event.endTime) {
+                  const startTime = formatLocalDate(
+                    event.startTime,
+                    'hh:mmA',
+                    this.timeZone,
+                  );
+                  const endTime = formatLocalDate(
+                    event.endTime,
+                    'hh:mmA z',
+                    this.timeZone,
+                  );
+                  openHouseStr += ` (${startTime} - ${endTime})`;
+                }
+              }
+              return openHouseStr;
+            })
+            .filter((str) => str.length)
+            .join(', ');
+        },
+      },
+      {
+        path: 'userAccounts',
+        label: 'Partners Who Have Access',
+        format: (val: User[]): string =>
+          val.map((user) => `${user.firstName} ${user.lastName}`).join(', '),
+      },
+    ];
+
+    return headers;
+  }
+
+  getUnitCsvHeaders(): CsvHeader[] {
+    return [
+      {
+        path: 'listing.id',
+        label: 'Listing Id',
+      },
+      {
+        path: 'listing.name',
+        label: 'Listing Name',
+      },
+      {
+        path: 'unit.number',
+        label: 'Unit Number',
+      },
+      {
+        path: 'unit.unitTypes.name',
+        label: 'Unit Type',
+      },
+      {
+        path: 'unit.numBathrooms',
+        label: 'Number of Bathrooms',
+      },
+      {
+        path: 'unit.floor',
+        label: 'Unit Floor',
+        format: this.hideZero,
+      },
+      {
+        path: 'unit.sqFeet',
+        label: 'Square Footage',
+      },
+      {
+        path: 'unit.minOccupancy',
+        label: 'Minimum Occupancy',
+        format: this.hideZero,
+      },
+      {
+        path: 'unit.maxOccupancy',
+        label: 'Max Occupancy',
+        format: this.hideZero,
+      },
+      {
+        path: 'unit.amiChart.name',
+        label: 'AMI Chart',
+      },
+      {
+        path: 'unit.amiChart.items.0.percentOfAmi',
+        label: 'AMI Level',
+      },
+      {
+        path: 'unit',
+        label: 'Rent Type',
+        format: (val: Unit) =>
+          !isEmpty(val.monthlyRentAsPercentOfIncome)
+            ? '% of income'
+            : !isEmpty(val.monthlyRent)
+            ? 'Fixed amount'
+            : '',
+      },
+    ];
+  }
+
+  async authorizeCSVExport(user): Promise<void> {
```

Comment:
same as prev pr, since perms are in place we should be able to apply perms properly